Client/Frontend/InternalSchemeHandler/ErrorPageHelper.swift

```diff
@@ -0,0 +1,336 @@
+    case .cfurlErrorRequestBodyStreamExhausted: return "CFURLErrorRequestBodyStreamExhausted"
+    case .cfurlErrorFileDoesNotExist: return "CFURLErrorFileDoesNotExist"
+    case .cfurlErrorFileIsDirectory: return "CFURLErrorFileIsDirectory"
+    case .cfurlErrorNoPermissionsToReadFile: return "CFURLErrorNoPermissionsToReadFile"
+    case .cfurlErrorDataLengthExceedsMaximum: return "CFURLErrorDataLengthExceedsMaximum"
+    case .cfurlErrorSecureConnectionFailed: return "CFURLErrorSecureConnectionFailed"
+    case .cfurlErrorServerCertificateHasBadDate: return "CFURLErrorServerCertificateHasBadDate"
+    case .cfurlErrorServerCertificateUntrusted: return "CFURLErrorServerCertificateUntrusted"
+    case .cfurlErrorServerCertificateHasUnknownRoot: return "CFURLErrorServerCertificateHasUnknownRoot"
+    case .cfurlErrorServerCertificateNotYetValid: return "CFURLErrorServerCertificateNotYetValid"
+    case .cfurlErrorClientCertificateRejected: return "CFURLErrorClientCertificateRejected"
+    case .cfurlErrorClientCertificateRequired: return "CFURLErrorClientCertificateRequired"
+    case .cfurlErrorCannotLoadFromNetwork: return "CFURLErrorCannotLoadFromNetwork"
+    case .cfurlErrorCannotCreateFile: return "CFURLErrorCannotCreateFile"
+    case .cfurlErrorCannotOpenFile: return "CFURLErrorCannotOpenFile"
+    case .cfurlErrorCannotCloseFile: return "CFURLErrorCannotCloseFile"
+    case .cfurlErrorCannotWriteToFile: return "CFURLErrorCannotWriteToFile"
+    case .cfurlErrorCannotRemoveFile: return "CFURLErrorCannotRemoveFile"
+    case .cfurlErrorCannotMoveFile: return "CFURLErrorCannotMoveFile"
+    case .cfurlErrorDownloadDecodingFailedMidStream: return "CFURLErrorDownloadDecodingFailedMidStream"
+    case .cfurlErrorDownloadDecodingFailedToComplete: return "CFURLErrorDownloadDecodingFailedToComplete"
+
+    case .cfhttpCookieCannotParseCookieFile: return "CFHTTPCookieCannotParseCookieFile"
+    case .cfNetServiceErrorUnknown: return "CFNetServiceErrorUnknown"
+    case .cfNetServiceErrorCollision: return "CFNetServiceErrorCollision"
+    case .cfNetServiceErrorNotFound: return "CFNetServiceErrorNotFound"
+    case .cfNetServiceErrorInProgress: return "CFNetServiceErrorInProgress"
+    case .cfNetServiceErrorBadArgument: return "CFNetServiceErrorBadArgument"
+    case .cfNetServiceErrorCancel: return "CFNetServiceErrorCancel"
+    case .cfNetServiceErrorInvalid: return "CFNetServiceErrorInvalid"
+    case .cfNetServiceErrorTimeout: return "CFNetServiceErrorTimeout"
+    case .cfNetServiceErrorDNSServiceFailure: return "CFNetServiceErrorDNSServiceFailure"
+    default: return "Unknown"
+    }
+}
+
+class ErrorPageHandler: InternalSchemeResponse {
+    static let path = URL.errorPagePath
+
+    func response(forRequest request: URLRequest) -> (URLResponse, Data)? {
+        guard
+            // request.isPrivileged, -> TODO: lock this down, session restore can redirect to an error page
```

Comment:
Ugh, `request.isPrivileged` won't work here. A session restore can push an error page onto the history stack, so we just get a request that came directly from the webview. I'll file a follow-up bug to deal with this.